torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -768,6 +770,28 @@ std::shared_ptr<JitFuture> TensorPipeAgent::send(
                 clientPipe.pendingResponseMessage_.erase(it);
               }
 
+              // Remove entry from timeoutMap_ as well.
```

Comment:
We might want to do the same in the error branch, i.e., at lines 743-745. What happens there is that all pending messages for this outgoing pipe are instantly completed with error, and dropped from `clientPipe.pendingResponseMessage_`. To make it neater we could also remove them from the `messageIdToExpiryMap_` and the `timeoutMap_`.